node/src/components/chainspec_loader.rs

```diff
@@ -412,6 +453,42 @@ where
             Event::Request(ChainspecLoaderRequest::GetCurrentRunInfo(responder)) => {
                 responder.respond(self.get_current_run_info()).ignore()
             }
+            Event::Request(ChainspecLoaderRequest::GetChainspecRawBytes(responder)) => {
+                let chainspec_file_path = self
+                    .root_dir
+                    .join(dir_name_from_version(&self.chainspec.protocol_version()))
+                    .join(CHAINSPEC_NAME);
+
+                let chainspec_bytes = match fs::read(chainspec_file_path) {
```

Comment:
I wonder if we should actually read a file from `fs` here. Is it possible that chainspec of genesis accounts changes when the node is running? If not, consider loading it once and caching, so we don't need to do the roundtrip to the filesystem each time we handle the GetChainspecRawBytes request.

Reply:
No, so the genesis accounts file will only be present for the 1_0_0 PV for a given network. So one could imagine in a 1.2 PV of a given network there is only the chainspec bytes. So depending on the given PV we may or may not need the value